### ISSUE TITLE
Failed to start Elixir. using Ubuntu wsl on Windows 10

### ISSUE THREAD

**Issue by WolfDan** [JakeBecker/vscode-elixir-ls#16]
I have configured my system so that elixir runs only on Ubuntu (No elixir etc.. installed in Windows)

Elixir and Phoenix works fine from the bash console (and I also manage to work in cmd by using `bash.exe -c "elixir %*"` in a .bat file), but ElixirLS simply don't work with WSL, here the log:

```
Failed to start Elixir.
error: {error,{elixir,{"no such file or directory","elixir.app"}}}
[Info  - 11:27:58] Connection to server got closed. Server will restart.
```

There is any way to make this work correctly? :D 

Greetings

**Comment by JakeBecker** [comment]
Hm. For starters, this probably requires that this ticket be done first: https://github.com/JakeBecker/elixir-ls/issues/12

It's going to be tricky to make this work out-of-the-box, but maybe it's possible to let the user override the launch command so you can run it with `bash.exe`. I'll look into it after ^ that ticket is done.

**Comment by yaBliznyk** [comment]
Try to use doskey
I set macros and use in cmd
![image](https://user-images.githubusercontent.com/2345820/36071911-00500982-0f49-11e8-898c-74d2c212859a.png)
https://docs.microsoft.com/en-us/windows/console/console-aliases documentation

**Comment by yaBliznyk** [comment]
You can use it for aliases like nodejs, elixirc, elixir, mix
and set aliases like phx.server=bash.exe -c "iex -S mix phx.server" and other

**Comment by JakeBecker** [comment]
In Windows, the server is launched from a batch file using the `elixir` executable in your path. If you alias that as @yaBliznyk recommends, it will probably work. Please re-open if you can't get it working with this approach.

**Comment by AlejandroHuerta** [comment]
I haven't been able to get this working myself. I've added the alias to both powershell and cmd and made sure that they persist and load through sessions but the extension continues to throw an error for not being able to fine `elixir` on my path. Does the extension run `language_server.bat` through cmd or powershell? Or is it controlled by some setting?

Any help would be appreciated as getting NIFs working on windows is a real pain outside of the WSL.

**Comment by sleepiejohn** [comment]
I tried all of it, including changing the language_server.bat inside extensions, no luck, maybe because Im not that expert on bash and batch

**Comment by WolfDan** [comment]
To point it out, the only solution I found for it was to run Visual Code within WSL, and open your project there, it works just fine most of the time

On how to do it, check this https://gist.github.com/fdmnio/604c6b811939468fdad06e3fbba942ed

If you are working with nifs and want to make elixirls work inside windows natively, check my post on Elixir Forum

https://elixirforum.com/t/making-nifs-easier-to-work-with-on-windows-10-and-making-work-elixir-ls-too/17353

**Comment by AlejandroHuerta** [comment]
I attempted to look into this again. From what I can tell, the node.js child-process library doesn't seem to load in the doskey aliases. I haven't been able to figure out a way to make sure that happens. I then decided to simply modify the commands the extension used. So in `language_server.bat`, `extension.js`, and `debugger.bat` I simply changed `elixir` to `wsl elixir` which tells windows to direct the command to the default wsl. Unfortunately that doesn't seem to be enough as I've now gotten a different error which seems to indicate that the ElixirLS modules aren't being loaded.

>>>[Info  - 10:05:01 PM] Connection to server got closed. Server will restart.
** (UndefinedFunctionError) function ElixirLS.LanguageServer.CLI.main/0 is undefined (module ElixirLS.LanguageServer.CLI is not available)
    ElixirLS.LanguageServer.CLI.main()
    (stdlib) erl_eval.erl:680: :erl_eval.do_apply/6
    (elixir) lib/code.ex:232: Code.eval_string/3

Not sure where to go from here.
